cpp/ql/src/semmle/code/cpp/controlflow/Nullness.qll

```diff
@@ -277,7 +277,7 @@ predicate mayReturnNull(Function f) {
     ret.getEnclosingFunction() = f
   )
   or
-  exists(ReturnStmt ret, Expr returned, ControlFlowNode init, LocalScopeVariable v |
+  exists(ReturnStmt ret, Expr returned, ControlFlowNode init, StackVariable v |
```

Comment:
> Hopefully
this leads to an improved nullness analysis since it avoids treating
`static T *x = nullptr;` the same as `static T *x; x = nullptr;`.

On the flip side, this library seems to be tracking 'possibly null' values rather than 'definitely null' values, and the value of `x` after `static T *x = nullptr;` _is_ possibly null.

Reply:
Good point. Perhaps the use of `LocalScopeVariable` in this library could instead be expanded to `NonFieldVariable` (if we had such a class)? In any case, the easiest thing to do right now is probably to change it back to `LocalScopeVariable`.

Reply:
Commit reverted.